tests/zfs-tests/tests/functional/bclone/bclone_common.kshlib

```diff
@@ -0,0 +1,266 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or https://opensource.org/licenses/CDDL-1.0.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2023 by Pawel Jakub Dawidek
+#
+
```

Comment:
A generic `verify_block_cloning` check would be nice to have.  You'll need to call this in all of the new tests.

```ksh
function verify_block_cloning
{
        if is_linux && [[ $(linux_version) -lt $(linux_version "4.5") ]]; then
                log_unsupported "copy_file_range not available before Linux 4.5"
        fi
}
```

You'll also need to add all the new tests to this block in `./test-runner/bin/zts-report.py.in`

```py
elif sys.platform.startswith('linux'):
    maybe.update({
```